src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java

```diff
@@ -90,4 +91,16 @@ List<HearingEntity> findByCourtCodeAndHearingDay(
         "and h.fk_court_case_id = cc.id group by h.hearing_id)",
         nativeQuery = true)
     Optional<List<HearingEntity>> findHearingsByCaseId(String caseId);
+
+    @Query(value = "select h.* as first_created  " +
+            "from hearing h  " +
+            "inner join hearing_day hday on hday.fk_hearing_id = h.id  " +
+            "where hday.hearing_day = :hearingDay and hday.court_code = :courtCode " +
+            "and h.deleted = false",
+            nativeQuery = true)
+    List<HearingEntity> filterHearings(
+            String courtCode,
+            LocalDate hearingDay,
+            String source
+    );
```

Comment:
I am thinking of building a dynamic query as we are going to have more filters.

Reply:
Yes, the work you did for pagination might come in handy here